pkg/urbit/jets/e/secp.c

```diff
@@ -208,3 +208,90 @@ u3we_make(u3_noun cor)
     return u3l_punt("secp-make", _cqes_make(has, prv));
   }
 }
+
+/* create a schnorr signature
+*/
+static u3_weak
+_cqes_sosi(u3_atom sk, u3_atom m, u3_atom a)
+{
+  c3_y key_y[32];
+  c3_y mes_y[32];
+  c3_y aux_y[32];
+
+  if ( (0 != u3r_bytes_fit(32, key_y, sk)) ||
```

Comment:
see my comment below in _cqes_sove

Reply:
Maybe you have tests and I'm missing them, but I would have expected to see them in pkg/arvo/tests/sys/zuse/crypto/secp256k1.hoon? Just a few cases (from an RFC if possible) is fine, please try things that are zero and things that are outside the size range for the inputs, too. The mis-sized values should crash, and the tests should pass with the jets enabled and disabled.

Reply:
Added test cases from BIP-340; I had tested against those manually before.